src/main/java/org/xstefank/model/Utils.java

```diff
@@ -27,31 +22,13 @@
     public static final String MESSAGE = "message";
     public static final String URL = "url";
 
-    public static String readTokenFromProperties(String dirName, String fileName) {
-        InputStream is = null;
-        File dir = new File(dirName);
-        File fileProp = new File(dir, fileName);
-
-        try {
-            is = new FileInputStream(fileProp);
-            Properties properties = new Properties();
-            properties.load(is);
+    private static PropertiesLoader propertiesLoader = PropertiesLoader.getInstance();
```

Comment:
I have nothing against making `PropertiesLoader` a singleton, but why is it necessary? you are using it only in this one place

Reply:
also I don't think there is a necessity for yet another class that just loads the properties, make it configuration loader that handles all of the configuration loading processing, or leave it here in `Utils` and just store the loaded properties in a field for further use; if you think the singleton wrapper you've implemented is better solution, please comment here your reasoning